.github/workflows/k8s-test.yml

```diff
@@ -0,0 +1,22 @@
+name: k8s test
+
+# Run on each new PR and each new push to existing PR
+on: [push, pull_request]
+
+jobs:
+  run-k8s-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: deploy kind cluster
+        run: |
+          tests/scripts/install-5nodes-kind-cluster.sh
```

Comment:
why do we need a cluster to run the unit test? Seems overkill. Also unit tests run already with the TravisCI job so I'm not if this code addition is needed.

Reply:
This is basically an integration test, I don't mind axing it.  What is the alternative to running a k8s cluster? How to test a way in which cluster reacts to accessing a secret that does not exist?   Could this functionality be tested with TravisCI? Please share your thoughts.

Reply:
We might be able to just use a unit test for this. I'm not sure how they are articulated but if we can mock a server response then we can test that a secret does not exist. Also, it looks like GitHub action needs to be approved by a maintainer and this has never been the case so far....

Reply:
Did we remove this code out in the recent version of this PR?